vrealize_operations_integration_sdk/validation/highlights.py

```diff
@@ -0,0 +1,97 @@
+                           long_collection_statistics.long_object_type_statistics.items()
+                           if stats.objects_growth_rate > 0]
+
+    highlights = Result()
+    if len(objects_with_growth):
+        for obj_type, growth in objects_with_growth:
+            highlights.with_warning(f"Objects of type '{obj_type}' displayed growth of {growth:.2f}% per hour.")
+
+    return highlights
+
+
+def highlight_metric_growth(long_collection_statistics: LongCollectionStatistics) -> Result:
+    objects_with_metrics_growth = [(object_type, stats.metrics_growth_rate) for
+                                   object_type, stats in
+                                   long_collection_statistics.long_object_type_statistics.items()
+                                   if stats.metrics_growth_rate > 0]
+
+    highlights = Result()
+    if len(objects_with_metrics_growth):
+        for obj_type, growth in objects_with_metrics_growth:
+            highlights.with_warning(f"Objects of type '{obj_type}' displayed metric growth of {growth:.2f}% per hour.")
+
+    return highlights
+
+
+def highlight_property_growth(long_collection_statistics: LongCollectionStatistics) -> Result:
+    objects_with_property_growth = [(object_type, stats.properties_growth_rate) for
+                                    object_type, stats in
+                                    long_collection_statistics.long_object_type_statistics.items()
+                                    if stats.properties_growth_rate > 0]
+
+    highlights = Result()
+    if len(objects_with_property_growth):
+        for obj_type, growth in objects_with_property_growth:
+            highlights.with_warning(f"Objects of type '{obj_type}' displayed property growth of {growth:.2f}% per hour")
+
+    return highlights
+
+
+def highlight_property_value_growth(long_collection_statistics: LongCollectionStatistics) -> Result:
+    objects_with_property_value_growth = [(object_type, stats.property_values_growth_rate) for
+                                          object_type, stats in
+                                          long_collection_statistics.long_object_type_statistics.items()
+                                          if stats.property_values_growth_rate > 0]
+
+    MAX_NUMBER_OF_PROPERTY_VALUES = min(32, long_collection_statistics.total_number_of_collections)
+    threshold = get_growth_rate(1, MAX_NUMBER_OF_PROPERTY_VALUES, long_collection_statistics.long_run_duration)
+
+    highlights = Result()
+    if len(objects_with_property_value_growth):
+        for obj_type, growth in objects_with_property_value_growth:
+            if growth > threshold:
+                highlights.with_error(
+                    f"Objects of type '{obj_type}' displayed excessive property value growth of {growth:.2f}% per hour.")
+            else:
+                highlights.with_warning(f"Objects of type '{obj_type}' displayed property value growth of {growth:.2f}% per hour.")
```

Comment:
One thing that we're missing with all of the warnings is some explanation. We show what the problem is, but not why it's a problem or what to look for to fix the issue. I think for anyone who is not as familiar with some of these issues as we are this will be very important context to provide.

Reply:
I like how these are reading. However, there are a few things that are incorrect in the explanations. Let's go through them one by one when you get some time.